org.eclipse.lsp4e/src/org/eclipse/lsp4e/outline/LSSymbolsContentProvider.java

```diff
@@ -139,6 +145,36 @@ public void documentAboutToBeChanged(DocumentEvent event) {
 		public void documentChanged(DocumentEvent event) {
 			refreshTreeContentFromLS();
 		}
+
+	}
+
+	private final class PreferencesChangedOutlineUpdater implements IPreferenceChangeListener, IOutlineUpdater {
+
+		@Override
+		public void install() {
+			IEclipsePreferences preferences = InstanceScope.INSTANCE.getNode(LanguageServerPlugin.PLUGIN_ID);
+			preferences.addPreferenceChangeListener(this);
+		}
+
+		@Override
+		public void uninstall() {
+			IEclipsePreferences preferences = InstanceScope.INSTANCE.getNode(LanguageServerPlugin.PLUGIN_ID);
+			preferences.removePreferenceChangeListener(this);
+		}
+
+		@Override
+		public void preferenceChange(PreferenceChangeEvent event) {
+			if (viewer != null
+					&& event.getKey().startsWith(CNFOutlinePage.HIDE_DOCUMENT_SYMBOL_KIND_PREFERENCE_PREFIX)) {
+				viewer.getControl().getDisplay().asyncExec(() -> {
+					if(viewer.getTree().isDisposed()) {
```

Comment:
Since there are connection problems, what do you think of simplifying it further and remove the return as in

```
					if(!viewer.getTree().isDisposed()) {
						viewer.refresh();
					}
```
?

Reply:
Yes, I'll change that, too.

Since rebasing on main branch, I get a lot of Java problems like the following. Do I have to change some configuration. I'm using JDK 21 with source compatibility set to 17.

![image](https://github.com/user-attachments/assets/882cba10-3d16-48ee-9bcb-b608f6ffdc3b)

Reply:
@sebthom , could you help with the errors? I suspect they are a follow-up of the last refactor.

Reply:
It seems, I had only to install the Maven Integration plug-ins that my IDE was missing. Maybe that was due to some pom-derived compile settings that were introduced with the latest commits in the main branch.